version: '3.8'

services:
  migrate_db:
    image: aaissaut/aaiss:backend
    build: ./backend
    command: ./manage.py migrate
    volumes:
      - ./backend/data:/backend-aaiss/data

  backend:
    depends_on:
      migrate_db:
        condition: service_completed_successfully
    image: aaissaut/aaiss:backend
    build: ./backend
    command: gunicorn aaiss_backend.wsgi:application --bind 0.0.0.0:6446
    volumes:
      - ./backend/data:/backend-aaiss/data
      - ./backend/static:/backend-aaiss/static
      - ./backend/media:/backend-aaiss/media
    expose:
      - 6446
    healthcheck:
      test: [ "CMD", "curl", "localhost:6446/admin" ]
      timeout: 2s
      retries: 10

  caddy:
    depends_on:
      backend:
        condition: service_healthy
    image: caddy:2.7.3
    volumes:
      - ./deployment/caddy:/etc/caddy
      - ./backend/static:/static
      - ./backend/media:/media
      - ./dist:/dist
      - caddy_data:/data/caddy
    ports:
      - "80:80"
      - "443:443"

volumes:
  caddy_data: